src/main/scala/viper/silver/plugin/standard/smoke/SmokeDetectionPlugin.scala

```diff
@@ -0,0 +1,215 @@
+package viper.silver.plugin.standard.smoke
+
+import fastparse.P
+import viper.silver.ast._
+import viper.silver.ast.utility.ViperStrategy
+import viper.silver.cfg.silver.SilverCfg
+import viper.silver.cfg.silver.SilverCfg.SilverBlock
+import viper.silver.cfg.utility.IdInfo
+import viper.silver.parser.FastParser
+import viper.silver.plugin.standard.refute.Refute
+import viper.silver.plugin.{ParserPluginTemplate, SilverPlugin}
+
+import scala.annotation.unused
+import scala.collection.mutable.ListBuffer
+
+class SmokeDetectionPlugin(@unused reporter: viper.silver.reporter.Reporter,
+                           @unused logger: ch.qos.logback.classic.Logger,
+                           @unused config: viper.silver.frontend.SilFrontendConfig,
+                           fp: FastParser) extends SilverPlugin with ParserPluginTemplate {
+
+  import fp.{FP, ParserExtension, keyword}
+
+  /** Keyword used to define `unreachable` statement. */
+  private val unreachableKeyword: String = "unreachable"
+
+  /** Parser for `unreachable` statements. */
+  def unreachable[$: P]: P[PUnreachable] =
+    FP(keyword(unreachableKeyword)).map { case (pos, _) => PUnreachable()(pos) }
+
+  /** Add `unreachable` to the parser. */
+  override def beforeParse(input: String, isImported: Boolean): String = {
+    // Add new keyword
+    ParserExtension.addNewKeywords(Set[String](unreachableKeyword))
+    // Add new parser to the precondition
+    ParserExtension.addNewStmtAtEnd(unreachable(_))
+    input
+  }
+
+  override def beforeVerify(input: Program): Program = {
+    var refuteId = 0
+
+    // Add `refute false` at end of each method body
+    val transformedMethods = input.methods.map(method => {
+      val body = if (method.body.isEmpty) Seq() else Seq(method.body.get)
+      val bodyWithRefute = Option(Seqn(body :+ refuteFalse(method.pos, refuteId), Seq())(method.pos))
```

Comment:
This seems problematic to me. Doesn't that turn abstract methods into non-abstract ones? E.g. if I have a method 
```
method m()
  requires P
  ensures Q
```
without a body, wouldn't this turn it into 
```
method m()
  requires P
  ensures Q
{ refute false }
```
which means I now have to prove that the postcondition holds after an empty body?

Reply:
You're right, good point! I fixed it by not touching the method `body` if it is `None`; if it is an empty `Seq` (which corresponds to an empty method body), then a `refute false` is inserted. ->e77a220f